Keas.Mvc/Controllers/Api/PeopleAdminController.cs

```diff
@@ -91,15 +91,41 @@ public async Task<IActionResult> Delete(int id)
             var person = await _context.People
                 .Where(x => x.Team.Slug == Team)
                 .Include(x => x.User)
-                .SingleAsync(x => x.Id == id);
+                .SingleAsync(x => x.Id == id);
+
+           
 
             var personActing = await _securityService.GetPerson(Team);
 
             if (personActing.UserId == person.UserId)
             {
                 ModelState.AddModelError("User", "Don't delete yourself.");
                 return BadRequest(ModelState);
-            }
+            }
+
+
+
+            if (await _context.AccessAssignments.AnyAsync(a => a.PersonId == id))
+            {
+                ModelState.AddModelError("AccessAssignments", "Remove Access Assignments first");
+                return BadRequest(ModelState);
+            }
+            if (await _context.KeySerialAssignments.AnyAsync(a => a.PersonId == id))
+            {
+                ModelState.AddModelError("KeySerialAssignments", "Remove Key Assignments first");
+                return BadRequest(ModelState);
+            }
+            if (await _context.EquipmentAssignments.AnyAsync(a => a.PersonId == id))
+            {
+                ModelState.AddModelError("EquipmentAssignments", "Remove Equipment Assignments first");
+                return BadRequest(ModelState);
+            }
+            if (await _context.WorkstationAssignments.AnyAsync(a => a.PersonId == id))
+            {
+                ModelState.AddModelError("WorkstationAssignments", "Remove Workstation Assignments first");
+                return BadRequest(ModelState);
+            }

```

Comment:
Maybe remove all the `return BadRequest(ModelState);` statements and instead increment a modelErrorCount, then return BadRequest if that error count > 0? That would give more info about what needs to be corrected. Otherwise it feels like a client developer would have to play api whack-a-mole.

Reply:
Or instead of using a modelErrorCount, just inspect ModelState for any errors.

Reply:
I figured it should be rare, and if the API consumer isn't checking this first then they will get the next error if there are multiple assignments.